tests/unit/sagemaker/workflow/test_lambda_step.py

```diff
@@ -0,0 +1,171 @@
+    boto_mock = Mock(name="boto_session", region_name="us-west-2")
+    session_mock = Mock(
+        name="sagemaker_session",
+        boto_session=boto_mock,
+        boto_region_name="us-west-2",
+        config=None,
+        local_mode=False,
+    )
+    return session_mock
+
+
+def test_lambda_step(sagemaker_session):
+    param = ParameterInteger(name="MyInt")
+    outputParam1 = LambdaOutput(output_name="output1", output_type=LambdaOutputTypeEnum.String)
+    outputParam2 = LambdaOutput(output_name="output2", output_type=LambdaOutputTypeEnum.Boolean)
+    lambda_step = LambdaStep(
+        name="MyLambdaStep",
+        depends_on=["TestStep"],
+        lambda_func=Lambda(
+            function_arn="arn:aws:lambda:us-west-2:123456789012:function:sagemaker_test_lambda",
+            session=sagemaker_session,
+        ),
+        inputs={"arg1": "foo", "arg2": 5, "arg3": param},
+        outputs=[outputParam1, outputParam2],
+    )
+    lambda_step.add_depends_on(["SecondTestStep"])
+    assert lambda_step.to_request() == {
+        "Name": "MyLambdaStep",
+        "Type": "Lambda",
+        "DependsOn": ["TestStep", "SecondTestStep"],
+        "FunctionArn": "arn:aws:lambda:us-west-2:123456789012:function:sagemaker_test_lambda",
+        "OutputParameters": [
+            {"OutputName": "output1", "OutputType": "String"},
+            {"OutputName": "output2", "OutputType": "Boolean"},
+        ],
+        "Arguments": {"arg1": "foo", "arg2": 5, "arg3": param},
+    }
+
+
+def test_lambda_step_output_expr(sagemaker_session):
+    param = ParameterInteger(name="MyInt")
+    outputParam1 = LambdaOutput(output_name="output1", output_type=LambdaOutputTypeEnum.String)
+    outputParam2 = LambdaOutput(output_name="output2", output_type=LambdaOutputTypeEnum.Boolean)
+    lambda_step = LambdaStep(
+        name="MyLambdaStep",
+        depends_on=["TestStep"],
+        lambda_func=Lambda(
+            function_arn="arn:aws:lambda:us-west-2:123456789012:function:sagemaker_test_lambda",
+            session=sagemaker_session,
+        ),
+        inputs={"arg1": "foo", "arg2": 5, "arg3": param},
+        outputs=[outputParam1, outputParam2],
+    )
+
+    assert lambda_step.properties.Outputs["output1"].expr == {
+        "Get": "Steps.MyLambdaStep.OutputParameters['output1']"
+    }
+    assert lambda_step.properties.Outputs["output2"].expr == {
+        "Get": "Steps.MyLambdaStep.OutputParameters['output2']"
+    }
+
+
+def test_pipeline_interpolates_lambda_outputs(sagemaker_session):
+    parameter = ParameterString("MyStr")
+    outputParam1 = LambdaOutput(output_name="output1", output_type=LambdaOutputTypeEnum.String)
+    outputParam2 = LambdaOutput(output_name="output2", output_type=LambdaOutputTypeEnum.String)
+    lambda_step1 = LambdaStep(
+        name="MyLambdaStep1",
+        depends_on=["TestStep"],
+        lambda_func=Lambda(
+            function_arn="arn:aws:lambda:us-west-2:123456789012:function:sagemaker_test_lambda",
```

Comment:
nit: For us as a whole team, we should in general rethink about refactoring and reusing duplicated arns/resource objects across tests

Reply:
Agree. This is just a dummy ARN though. It will create a pipeline which has that ARN in the functionArn field of Lambda step in the pipeline definition. We just need to verify if the pipeline gets created with the correct function arn. Staring a pipeline execution invokes the lambda function which will require actual resource, but that functionality is not tested here since no change in that behavior and the same has been tested in existing workflow tests

Reply:
That'a fair. 

There is some behavior here though that is specific to LambdaStep, for example the Outputs can only be of certain types. What happens if the Output generated is of another type? Do we throw an error?

Could you point me to existing tests that check this scenario

Reply:
Sure. Here are some of those. There are more
https://github.com/aws/sagemaker-python-sdk/blob/master/tests/integ/test_workflow.py#L476
https://github.com/aws/sagemaker-python-sdk/blob/master/tests/integ/test_workflow.py#L578
https://github.com/aws/sagemaker-python-sdk/blob/master/tests/integ/test_workflow.py#L675

Reply:
If the output generated is of different type, and is referenced in another step, tioga will throw an error while validation it. This will be an error while running the pipeline and pipeline execution will be in failed status. We don't have a test for this specific scenario in sdk since this is a pipeline execution error and we are not throwing any error in SDK

Reply:
Makes sense, thanks for clarifying!